releases/release-4.0.5.md

```diff
@@ -0,0 +1,94 @@
+    - Fix issue `pre_split_regions` table option doesn't work in the partition table. [#18837](https://github.com/pingcap/tidb/pull/18837)
+    - Fixed an issue that could cause a large transaction to be terminated prematurely. [#18813](https://github.com/pingcap/tidb/pull/18813)
+    - Fix incorrect collator when `getSignatureByPB` and remove unnecessary recover [#18735](https://github.com/pingcap/tidb/pull/18735)
+    - Fix a bug `getAutoIncrementID()` function does not consider the `tidb_snapshot` session variable, this bug may cause dumper tool fail with 'table not exist' error. [#18692](https://github.com/pingcap/tidb/pull/18692)
+    - Fix unknown column error for sql like `select a from t having t.a` [#18434](https://github.com/pingcap/tidb/pull/18434)
+    - Support the Action when memory exceed quota for TableReader Executor. [#18392](https://github.com/pingcap/tidb/pull/18392)
+    - Fix a panic on hash partition table when the query condition is that the partition column equals to a big number like 9223372036854775808 [#18186](https://github.com/pingcap/tidb/pull/18186)
+    - Refine the behavior of StrToInt and StrToFloat and support convert JSON to date, time and timestamp [#18159](https://github.com/pingcap/tidb/pull/18159)
+    - Fix the wrong behavior of char function. [#18122](https://github.com/pingcap/tidb/pull/18122)
+    - Ddl: fix admin repair table with range partition expr will parseInt fail [#17988](https://github.com/pingcap/tidb/pull/17988)
+    - Fix wrong behavior of set charset statement [#17289](https://github.com/pingcap/tidb/pull/17289)
+    - Fix a bug caused by the wrong collation setting which leads to the wrong result of collation function. [#17231](https://github.com/pingcap/tidb/pull/17231)
+    - Fix STR_TO_DATE's handling for format token '%r', '%h' [#18727](https://github.com/pingcap/tidb/pull/18727)
+    - Fix issues of TiDB version information formation doesn't consistent with PD/TiKV in cluster_info table. [#18413](https://github.com/pingcap/tidb/pull/18413)
+
++ PD
+
+    - Fix the bug that TSO request may fail at the time of leader changing. [#2666](https://github.com/pingcap/pd/pull/2666)
+    - Fix the issue that when enabling placement rules, sometimes region replicas cannot schedule to optimal [#2720](https://github.com/pingcap/pd/pull/2720)
+
+    
++ TiFlash
+
+    - Fix the issue that TiFlash cannot start normally after upgrading from an old version if the name of the database or table contains special characters. [#971](https://github.com/pingcap/tics/pull/971)
+    - Fix the issue that TiFlash can not exit if any exception is thrown during initialization [#953](https://github.com/pingcap/tics/pull/953)
+    
++ TiKV
+
+    - Fix memory leak during scheduling [#8357](https://github.com/tikv/tikv/pull/8357)
+    - Speed up leader election when hibernate region is enabled [#8292](https://github.com/tikv/tikv/pull/8292)
```

Comment:
```suggestion
    - Speed up leader election when hibernate region is enabled [#8292](https://github.com/tikv/tikv/pull/8292)

# TiCDC

## New Features

- Support Kafka SSL connection [#764](https://github.com/pingcap/ticdc/pull/764)
- Support to output old value [#708](https://github.com/pingcap/ticdc/pull/708)

## Improvements

- Optimize get tso frequency [#801](https://github.com/pingcap/ticdc/pull/801)

## Bug Fixes

- Fix a issue that failed changefeed can't be removed [#782](https://github.com/pingcap/ticdc/pull/782)
- Fix invalid delete evenets by selecting one unique index as the handle index [#787](https://github.com/pingcap/ticdc/pull/787)
- Fix the bug that GC safepoint will be forwarded beyond the checkpoint of stopped changefeed [#797](https://github.com/pingcap/ticdc/pull/797)

# BR

## Improvements

- Optimize some log [#428](https://github.com/pingcap/br/pull/428)

# Dumpling

## Improvements

- Release FTWRL after connections are created to reduce lock time for mysql [#121](https://github.com/pingcap/dumpling/pull/121)

# TiDB-Lightning

## Improvements

- Optimize some log [#352](https://github.com/pingcap/tidb-lightning/pull/352)

## Bug Fixes

- Fix syntax error on empty binary/hex literals when using TiDB backend [#357](https://github.com/pingcap/tidb-lightning/pull/357)

```

Reply:
@overvenus Please add these notes to the `## Improvements`, `## Bug Fixes`, and `New Features` sections respectively.

Reply:
ticdc Features need add two：
Support the column flags [#796](https://github.com/pingcap/ticdc/pull/796)
Support to output previous table info in ddl events [#799](https://github.com/pingcap/ticdc/pull/799)

Reply:
@uglyengineer @overvenus Added in https://github.com/pingcap/docs/pull/3541/commits/7fc428457585806f8ae17121e4ebddf1e5a8e3e0